### ISSUE TITLE
[ENHANCEMENT] Skip Tainting for Externally Managed Nodes

### ISSUE THREAD

**Issue by mattmattox** [harvester/harvester#6305]
**Is your enhancement related to a problem? Please describe.**

When you join a node to a cluster running with the Harvester Cloud Provider that is not a Harvester node, it automatically gets tainted with `node.cloudprovider.kubernetes.io/uninitialized=true:NoSchedule`. This taint is intended to prevent scheduling on nodes that Harvester does not fully initialize. However, this behavior can cause issues in mixed environments when mixing VMs and Physical servers.

Note: if you 

**Describe the solution you'd like**

I would like to enhance the existing VirtualMachineInstance controller to support a new label, `harvesterhci.io/external-managed-node=true`, indicating nodes managed externally (i.e., not by Harvester). When this label is present on a node, the controller should skip the tainting process for that node, allowing Harvester and non-Harvester nodes to coexist in the same cluster without conflicts.

**Describe alternatives you've considered**

My workaround is to have a script that watches for taint to be added to these nodes and remove it.

**Additional context**

I'm using this for GPU nodes.

**Comment by bk201** [comment]
Hi @mattmattox, thanks for reporting.
>When you join a node to a cluster running with the Harvester Cloud Provider

Do you mean you create a guest cluster via Rancher (all Kubernetes nodes are VM nodes) and manually join another worker node to the cluster?

**Comment by mattmattox** [comment]
> Hi @mattmattox, thanks for reporting.
> 
> > When you join a node to a cluster running with the Harvester Cloud Provider
> 
> Do you mean you create a guest cluster via Rancher (all Kubernetes nodes are VM nodes) and manually join another worker node to the cluster?

Yes, in my case, I'm deploying an RKE2 cluster from Rancher using Harvester as my node driver. Then, I'm joining some physical nodes to this downstream cluster.

These are Dell R720xd with K80 GPUs, and they don't like the PCI passthrough and don't support vGPU.
